docs/Trendmark.md

```diff
@@ -0,0 +1,149 @@
+interpret the content of the watermark correctly as long as the watermark follows this
+specification. The table below lists all specified tags and explains their purpose. The tags are
+counted upwards from 0 for watermarks that add additional content and downwards from fe for
+watermarks that change the content itself (e.g. compression).
+
+| Tag (in hex) | Class name | Meaning |
+| -- | -- | -- |
+| 00 | RawWatermark | raw bytes without special encoding |
+| 01 | SizedWatermark | size + raw bytes |
+| 02 | CRC32Watermark | CRC32 checksum + raw bytes |
+| 03 | SizedCRC32Watermark | size + CRC32 checksum + raw bytes |
+| 04 | SHA3256Watermark | SHA3-256 hash + raw bytes |
+| 05 | SizedSHA3256Watermark | size + SHA3-256 hash + raw bytes |
+| -- | -- | -- |
+| fe | CompressedRawWatermark | compressed bytes |
+| fd | CompressedSizedWatermark | size + compressed bytes |
+| fc | CompressedCRC32Watermark | CRC32 checksum + compressed bytes |
+| fb | CompressedSizedCRC32Watermark | size + CRC32 checksum + compressed bytes |
+| fa | CompressedSHA3256Watermark | SHA3-256 hash + compressed bytes |
+| f9 | CompressedSizedSHA3256Watermark | size + SHA3-256 hash + compressed bytes |
+| -- | -- | -- |
+| ff | Custom | Reserved for custom Trendmark implementations |
+
+## Details
+In the following each watermark is specified with the text ``Lorem Ipsum`` encoded in UTF-8 bytes
+(4c 6f 72 65 6d 20 69 70 73 75 6d) as example watermark content. All values are in hexadecimal
+format.
+The compressed variants use DEFLATE as compression algorithm (see RFC 1951).
+Compressing the example watermark content results in the bytes (f3 c9 2f 4a cd 55 f0 2c 28 2e cd 05
+00), which is larger because the example content is too small to be effectively compressed.
+Only the content is compressed, potentially allowing to use the additional information to increase 
+the watermark robustness.
+
+### RawWatermark
+
+| tag | raw bytes |
+| -- | -- |
+| 00 | 4c 6f 72 65 6d 20 69 70 73 75 6d |
+
+### SizedWatermark
+The size is calculated over the entire watermark.
+
+| tag | size in 32 bits little-endian | raw bytes |
+| -- | -- | -- |
+| 01 | 10 00 00 00 | 4c 6f 72 65 6d 20 69 70 73 75 6d |
+
+### CRC32Watermark
+The CRC32 checksum is calculated over the entire watermark, replacing the bytes containing the
+checksum with null bytes.
+
+| tag | CRC32 checksum little-endian | raw bytes |
+| -- | -- | -- |
+| 02 | 87 0b 16 35 | 4c 6f 72 65 6d 20 49 70 73 75 6d |
+
+### SizedCRC32Watermark
+The size and CRC32 checksum are calculated over the entire watermark, replacing the bytes containing
+the checksum with null bytes.
+
+| tag | size in 32 bits little-endian | CRC32 checksum little-endian | raw bytes |
+| -- | -- | -- | -- |
+| 03 | 14 00 00 00 | 1e 85 5b 04 | 4c 6f 72 65 6d 20 49 70 73 75 6d |
+
+### SHA3256Watermark
+The SHA3-256 hash is calculated over the entire watermark, replacing the bytes containing the hash
```

Comment:
Add a link to SHA3-256 so people can read further information if they aren't familiar with it.